extensions/bitwarden/src/search.tsx

```diff
@@ -0,0 +1,191 @@
+  CopyToClipboardAction,
+  getPreferenceValues,
+} from "@raycast/api";
+import { Item, Folder } from "./types";
+import { useEffect, useState } from "react";
+import treeify from "treeify";
+import { filterNullishPropertiesFromObject, codeBlock, titleCase } from "./utils";
+import { useBitwarden as useSessionToken } from "./hooks";
+import { TroubleshootingGuide, UnlockForm } from "./components";
+import { existsSync } from "fs";
+import { BitwardenApi } from "./api";
+import { dirname } from "path";
+
+const { cliPath, clientId, clientSecret, fetchFavicons } = getPreferenceValues();
+process.env.PATH = dirname(cliPath)
+const bitwardenApi = new BitwardenApi(clientId, clientSecret);
+
+export default function ListCommand(): JSX.Element {
+  if (!existsSync(cliPath)) {
+    return <TroubleshootingGuide />;
+  }
+
+  const [sessionToken, setSessionToken] = useSessionToken(bitwardenApi);
+
+  if (sessionToken === null) return <UnlockForm setSessionToken={setSessionToken} bitwardenApi={bitwardenApi} />;
+
+  return <ItemList bitwardenApi={bitwardenApi} sessionToken={sessionToken}/>;
+}
+
+function ItemList(props: { bitwardenApi: BitwardenApi; sessionToken: string | undefined }) {
+  const { bitwardenApi, sessionToken } = props;
+  const [state, setState] = useState<{ folders: Folder[]; items: Item[] }>();
+
+  const folderMap: Record<string, Folder> = {};
+  for (const folder of state?.folders || []) {
+    if (folder.id) folderMap[folder.id] = folder;
+  }
+
+  async function loadItems(sessionToken: string) {
+    try {
+      const [items, folders] = await Promise.all([
+        bitwardenApi.listItems<Item>("items", sessionToken),
+        bitwardenApi.listItems<Folder>("folders", sessionToken),
+      ]);
+
+      setState({ items, folders });
+    } catch (error) {
+      showToast(ToastStyle.Failure, "Failed to search vault", "Vault is locked");
+    }
+  }
+
+  useEffect(() => {
+    if (sessionToken) loadItems(sessionToken);
+  }, [sessionToken]);
+
+  return (
+    <List isLoading={typeof state === "undefined"}>
+      {state?.items.map((item) => (
+        <ItemListItem
+          key={item.id}
+          item={item}
+          folder={item.folderId ? folderMap[item.folderId] : undefined}
+          refreshItems={async () => {
+            if (sessionToken) {
+              const toast = await showToast(ToastStyle.Animated, "Syncing Items...");
+              await bitwardenApi.syncItems(sessionToken)
+              await loadItems(sessionToken);
+              await toast.hide();
+            }
+          }}
+        />
+      ))}
+    </List>
+  );
+}
+
+function ItemListItem(props: { item: Item; folder: Folder | undefined; refreshItems?: () => void }) {
+  const { item, folder, refreshItems } = props;
+  const { type, name, notes, identity, login, secureNote, fields, passwordHistory, card } = item;
+  const accessoryIcons = [];
+  if (folder) accessoryIcons.push(`📂 ${folder.name}`);
+  if (item.favorite) accessoryIcons.push(`⭐️`);
```

Comment:
```suggestion

  if (item.favorite) { accessoryIcons.push(`⭐️`); }
  if (folder) { accessoryIcons.push(`📂 ${folder.name}`); }
```
In addition to adding the `{ }` (:trollface:), I've also re-ordered these. Since the length of the folder's name varies between items, it would make the favorite icon appear misaligned in the list.

Reply:
I'm also a Python dev, so it's no big deal! Also, I didn't realize the icons were rendered in reverse order, so I think the order of your `push` calls is what we want 🙄